### PR TITLE
[master] Porting #49494 to master

### PR DESCRIPTION
**Pull request by garethgreenaway**
Porting #49494 to master

### CODE REVIEW

File: salt/renderers/gpg.py [hunk 1]
```diff
@@ -37,6 +37,56 @@
...
+    gpg --import /path/to/pubkey.gpg
+
+If the salt master runs as normal user, become this user before importing the
+keys. The default target dir will be ``~/.gnupg``. This can be overridden by
+the ``--homedir`` option. The keys must be at least readable for the runuser.
+
+Adjust trust level of imported keys
+***********************************
+
+In some cases, importing existing keys may not be enough an the trust level of
```
Comment (bryceml) [inline salt/renderers/gpg.py]: s/an/and/

File: salt/renderers/gpg.py [hunk 2]
```diff
@@ -37,6 +37,56 @@
...
 
+Import keys to a master
+************************
+
+If the keys already exist and need to be imported to the salt master, run the
+following to import them.
+
+.. code-block:: bash
+
+    gpg --import /path/to/private.key
```
Comment (bryceml) [inline salt/renderers/gpg.py]: the `--homedir /etc/salt/gpgkeys` part should probably be mentioned here.

File: salt/renderers/gpg.py [hunk 3]
```diff
@@ -37,6 +37,56 @@
...
+***********************************
+
+In some cases, importing existing keys may not be enough an the trust level of
+the key needs to be adjusted. This can be done by editing the key. The ``key_id``
+and the actual trust level of the key can be seen by listing the already imported
+keys.
+
+.. code-block:: bash
+
+    gpg --list-keys
```
Comment (bryceml) [inline salt/renderers/gpg.py]: I would add `--homedir /etc/salt/gpgkeys` here as well

File: salt/renderers/gpg.py [hunk 4]
```diff
@@ -37,6 +37,56 @@
...
+    gpg --list-secret-keys
+
+If the trust-level is not ``ultimate`` it needs to be changed by running
+
+.. code-block:: bash
+
+    gpg --edit-key <key_id>
```
Comment (bryceml) [inline salt/renderers/gpg.py]: I would add `--homedir /etc/salt/gpgkeys` here as well

File: salt/renderers/gpg.py [hunk 5]
```diff
@@ -37,6 +37,56 @@
...
+Enable usage of GPG keys on the master
+**************************************
+
+Generating or importing the keys is not enough to activate the ability to decrypt
+the pillars, especially if the keys are generated/imported in a non-standard dir.
+
+To enable the keys on the salt-master, the following needs to be added to the
+masters configuration.
+
+.. code-block:: bash
```
Comment (bryceml) [inline salt/renderers/gpg.py]: this should be yaml

File: salt/renderers/gpg.py (hunk 2 above)
Comment (waynew) [reply salt/renderers/gpg.py]:
```suggestion
    gpg  --homedir /etc/salt/gpgkeys --import /path/to/private.key
```

File: salt/renderers/gpg.py (hunk 1 above)
Comment (waynew) [reply salt/renderers/gpg.py]:
```suggestion
In some cases, importing existing keys may not be enough and the trust level of
```

File: salt/renderers/gpg.py (hunk 3 above)
Comment (waynew) [reply salt/renderers/gpg.py]:
```suggestion
    gpg  --homedir /etc/salt/gpgkeys --list-keys
```

File: salt/renderers/gpg.py (hunk 4 above)
Comment (waynew) [reply salt/renderers/gpg.py]:
```suggestion
    gpg --homedir /etc/salt/gpgkeys --edit-key <key_id>
```

File: salt/renderers/gpg.py (hunk 5 above)
Comment (waynew) [reply salt/renderers/gpg.py]:
```suggestion
.. code-block:: yaml
```

**Review by waynew — CHANGES_REQUESTED** [review]

**Review by waynew — APPROVED** [review]

**Pull request merged**